extensions-core/multi-stage-query/src/main/java/org/apache/druid/msq/querykit/DataSourcePlan.java

```diff
@@ -170,6 +171,18 @@ public static DataSourcePlan forDataSource(
           minStageNumber,
           broadcast
       );
+    } else if (dataSource instanceof UnionDataSource) {
```

Comment:
Please update the MSQ known issues and the docs where ever we are calling union all as unsupported in MSQ.